### DISCUSSION TITLE
getCurrentUVFromDayMax

### DISCUSSION THREAD

**Discussion by chunshek** [breezy-weather/breezy-weather#1144] [Ideas]
### Describe your idea

I suggest replacing the `getCurrentUVFromDayMax` formula with a more attenuated model that is a better fit for the diurnal variation of UV radiation.

### Other details

I live in the subtropics, where the UV Index gets very high (in excess of 11) at noon during the summer months. It is usually not too bad in the early morning or late afternoon. However, I notice that Breezy Weather would overestimate the UV Index during those early and late hours.

Upon investigating, I found out the function `getCurrentUVFromDayMax` uses this formula to model the UV radiation from the forecast maximum: https://www.desmos.com/calculator/lna7dco4zi

<img width="1345" alt="Screenshot 2024-06-16 at 09 12 21" src="https://github.com/breezy-weather/breezy-weather/assets/2017712/34d91b1e-f133-4608-845d-6bfa947a32b7">

While this curve is a valid model for global solar radiation, it severely overestimates the UV radiation in the early and late daytime hours. Compare the graph above against the following graph from the U.S. National Weather Service. You can see that it models the red curve (global radiation) rather than the blue curve (UV radiation).

<img width="731" alt="Screenshot 2024-06-16 at 09 21 12" src="https://github.com/breezy-weather/breezy-weather/assets/2017712/f459383f-52bd-4962-8858-aeb1cccfdeda">

> Notice how UV radiation is much more attenuated than the global at high solar zenith angles in the early morning and late afternoon. This is because the two components which make up UV and all other forms of radiation; the direct and the diffuse, are both greatly affected at these low sun angles.
>
> Source: https://www.cpc.ncep.noaa.gov/products/stratosphere/uv_index/uv_diurnal.shtml

I would like to suggest modeling the UV index using a different formula. By making a variable _θ_ = −π at sunrise, _θ_ = 0 at local noon, and _θ_ = +π at sunset, we can calculate cos(_θ_), normalize the output to a range between 0 and 1, multiply it by `dayMaxUV` and generate a much more fitting model, like the green curve below:

<img width="721" alt="Screenshot 2024-06-16 at 09 27 29" src="https://github.com/breezy-weather/breezy-weather/assets/2017712/44664396-819e-4255-8958-bcf39709f036">

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by chunshek** [comment]
My source is in [this branch](https://github.com/breezy-weather/breezy-weather/commit/3eb709abf2de7077de9dcc93a0f728a19536dfe4).

**Comment by papjul** [comment]
Available since v6.0.1-alpha
